linux/LibSensors.c

```diff
@@ -92,13 +113,47 @@ int LibSensors_getCPUTemperatures(CPUData* cpus, int cpuCount) {
             if (r != 0)
                continue;
 
-            cpus[feature->number].temperature = temp;
-            tempCount++;
+            cpus[tempId].temperature = temp;
+            if (tempId > 0)
+               coreTempCount++;
          }
       }
    }
 
-   return tempCount;
+   const double packageTemp = cpus[0].temperature;
+
+   /* Only package temperature - copy to all cpus */
+   if (coreTempCount == 0 && !isnan(packageTemp)) {
+      for (unsigned int i = 1; i <= cpuCount; i++)
+         cpus[i].temperature = packageTemp;
+
+      return;
+   }
+
+   /* No package temperature - set to max core temperature */
+   if (isnan(packageTemp) && coreTempCount != 0) {
+      double maxTemp = NAN;
+      for (unsigned int i = 1; i <= cpuCount; i++) {
+         const double coreTemp = cpus[i].temperature;
+         if (isnan(coreTemp))
+            continue;
+
+         maxTemp = MAXIMUM(maxTemp, coreTemp);
+      }
+
+      cpus[0].temperature = maxTemp;
+
+      return;
```

Comment:
Leave this `return` out and the next condition will handle filling up the remaining cores for HT/SMT.

Reply:
Thanks, I missed that case.